File: pinyougou-web/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/GoodsController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.common.pojo.PageResult;
import com.pinyougou.pojo.Goods;

import com.pinyougou.service.GoodsService;
import jdk.nashorn.internal.runtime.logging.Logger;

import org.apache.commons.lang3.StringUtils;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/goods")
public class GoodsController {

    @Reference(timeout = 10000)
    private GoodsService goodsService;

    @GetMapping("/findByPage")
    public PageResult findByPage(Integer page, Integer rows, Goods goods){
        try {
            goods.setAuditStatus("0");
            if (StringUtils.isNoneBlank(goods.getGoodsName())){
                goods.setGoodsName(new String(goods.getGoodsName()
                .getBytes("ISO8859-1"),"utf-8"));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return goodsService.findByPage(page,rows,goods);
    }

    @GetMapping("/checkPass")
    public boolean checkPass(Long[] ids,String status){
        try {
            goodsService.checkPass(ids,status);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    @GetMapping("/deleteGoods")
    public boolean deleteGoods(Long[] ids){
        try {
            goodsService.deleteGoods(ids);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: pinyougou-web/pinyougou-manager-web/src/main/webapp/js/controller/goodsController.js
app.controller("goodsController", function ($scope,$controller, baseService) {
    $controller('baseController', {$scope: $scope});
    $scope.status = ['未审核','已审核','审核未通过','关闭'];
    /** 定义搜索对象 */
    $scope.searchEntity = {};

    $scope.search = function (page, rows) {
        baseService.findByPage("/goods/findByPage", page, rows, $scope.searchEntity)
            .then(function (value) {
                $scope.dataList = value.data.list;
                $scope.paginationConf.totalItems = value.data.total;
            });
    };

    $scope.checkPass=function (status) {
        if ($scope.ids.length>0){

        baseService.sendGet("/goods/checkPass?ids="+$scope.ids+"&" +
            "status="+status)
            .then(function (value) {
                if (value.data){
               alert("操作成功");
                $scope.ids=[];
                $scope.reload();
                }else {
                    alert("操作出错，再来一次");
                }
            });
        }else {
            alert("请选择要操作的商品");
        }
    };

    $scope.delete=function () {
        if ($scope.ids.length>0){
            baseService.sendGet("/goods/deleteGoods?ids="+$scope.ids)
                .then(function (value) {
                   if (value.data){
                       alert("操作成功");
                       $scope.ids=[];
                       $scope.reload();
                   }else {
                       alert("操作失败");
                   }
                });
        }else {
            alert("请选择你要删除的商品");
        }
    };

});